source/Camera/InfoUtilOther/C++Halcon/HalconConnectUsingGenICam/HalconConnectUsingGenICam.cpp

```diff
@@ -0,0 +1,102 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string filename)
+{
+	const HalconCpp::HString filetype{ "ply" };
+	const HalconCpp::HString filenameCstring{ filename.c_str() };
+	const HalconCpp::HString genParamName{ "invert_normals" };
+	const HalconCpp::HString genParamValue{ "false" };
+	model.WriteObjectModel3d(filetype, filenameCstring, genParamName, genParamValue);
+}
+
+
+HalconCpp::HObjectModel3D setColorInObjectModel3D(HalconCpp::HImage RGB, HalconCpp::HObjectModel3D pcHalcon) {
```

Comment:
There is no need to create a new `HalconCpp::HObjectModel3D` object on the return. The return can be `void` but pass `HalconCpp::HObjectModel3D` object by reference, i.e. `void setColorInObjectModel3D(HalconCpp::HObjectModel3D &objectModel3D, HalconCpp::HImage &RGB)`

Then just keep:

`objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);`...

Reply:
Also, RGB can be const as it is not being changed in the function: `void setColorInObjectModel3D(HalconCpp::HObjectModel3D &objectModel3D, const HalconCpp::HImage &RGB)`

Reply:
I agree:
- If you essentially modify a given point cloud, just take it as (non-const) reference and return void.
- RGB should be const reference.